scripts/crew/CrewMember.gd

```diff
@@ -150,36 +143,26 @@ func _on_idle_state_physics_processing(delta: float) -> void:
 
 func _on_timer_timeout() -> void:
 	set_current_animation()
-	switch_visibility()
 	animation_player.play(current_animation)
 
 func start_assigning() -> void:
 	gui_open_perm = false
-	state_manager.send_event("assigning")
+	state_manager.send_event(&"assign")
 
 func assign(room: Room, center: Vector2) -> void:
-	state_manager.send_event("assigned")
+	state_manager.send_event(&"assigned")
 	print("assigned to room ", room.id)
 	set_movement_target(center)
+	state_manager.set_expression_property(&"assignment", &"work")
+	state_manager.send_event(&"walk")
 
-func _on_assignment_state_physics_processing(delta: float) -> void:
-	if !navigation_agent.is_navigation_finished():
-		set_rounded_direction()
-		velocity = velocity.lerp(current_direction.normalized() * speed, acceleration * delta)
-		move_and_slide()
 
-func switch_visibility() -> void:
-	sprite_idle.hide()
+func _on_working_state_entered() -> void:
+	print("working...")
+	animation_state = AnimationState.IDLE
 	sprite_walk.hide()
-	#sprite_work.hide()
-	#sprite_chat.hide()
-
-	match current_state:
-		State.IDLE:
-			sprite_idle.show()
-		State.WALK:
-			sprite_walk.show()
-		#State.WORK:
-			#sprite_work.show()
-		#State.CHAT:
-			#sprite_chat.show()
+	sprite_idle.show()
```

Comment:
Nice. Though I thought we could have a switch function like `switch_visibility` that toggles hide() on all animations that do not match the current animation state. Just so we do not have to add more `sprite_new_state.hide()` in multiple places if new states are added in the future.